precompiles/common/IICS20Authorization.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17;
+
+import "Types.sol";
+
+/// @author Evmos Team
+/// @title Authorization Interface
+/// @dev The interface through which solidity contracts will interact with smart contract approvals.
+interface IICS20Authorization {
+    /// @dev Emitted when an ICS-20 transfer authorization is granted.
+    /// @param grantee The address of the grantee.
+    /// @param granter The address of the granter.
+    /// @param allocations An array of Allocation authorized with this grant.
+    event IBCTransferAuthorization(
+        address indexed grantee,
+        address indexed granter,
+        ICS20Allocation[] allocations
+    );
+
+    /// @dev Approves IBC transfer with a specific amount of tokens.
+    /// @param grantee The address for which the transfer authorization is granted.
+    /// @param allocations An array of Allocation for the authorization.
+    function approve(address grantee, ICS20Allocation[] calldata allocations) external returns (bool approved);
+
+    /// @dev Revokes IBC transfer authorization for a specific grantee.
+    /// @param grantee The address for which the transfer authorization will be revoked.
+    function revoke(address grantee) external returns (bool revoked);
+
+    /// @dev Increase the allowance of a given grantee by a specific amount of tokens for IBC transfer methods.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
```

Comment:
Is the contract of an address only or a general address?

Reply:
It's a contract address and takes the origin as the `granter`